src/ucp/wireup/wireup_cm.c

```diff
@@ -670,8 +715,9 @@ static ssize_t ucp_cm_server_priv_pack_cb(void *arg, const char *dev_name,
 
     tl_bitmap = ucp_ep_get_tl_bitmap(ep);
     /* make sure that all lanes are created on correct device */
+    ucs_assert(pack_handle->field_mask & UCT_SOCKADDR_PRIV_DATA_PACK_HANDLE_DEVICE_NAME);
```

Comment:
why on server it's assertion, but on client it's ucs_error()? should be same in both cases (BTW i think assertion is enough)

Reply:
it depends on how the field is used. in this case, it's used only for the assertion in the next line so we don't really need to print a message to the user since we are not using the field, it's not really required. in the other cases where a ucs_error is printed, the fields are mandatory since we use them in the code so if they are missing the user should be notified.